lua/telescope/make_entry.lua

```diff
@@ -655,12 +655,16 @@ function make_entry.gen_from_ctags(opts)
     }
   end
 
-  return function(line)
-    if line == '' or line:sub(1, 1) == '!' then
+  return function(result)
+    if result.line == '' or result.line:sub(1, 1) == '!' then
       return nil
     end
 
-    local tag, file, scode = string.match(line, '([^\t]+)\t([^\t]+)\t/^\t?(.*)/;"\t+.*')
+    local tag, file, scode = string.match(result.line, '([^\t]+)\t([^\t]+)\t/^\t?(.*)/;"\t+.*')
+
+    if not path.exists(file) then
+      file = path.join(result.tags_directory, file)
+    end
```

Comment:
We should not do this for every line. Hitting the filesystem for every entry.
We just have to do this when displaying the preview and when opening a entry.
So you wanna add a `pre = function()` on action which will update the filename. (Pre will run first then `_goto_file_selection` and then post, which does the jump). Take a look at `attach_mappings` in `lua/telescope/builtin/files.lua:216-218`, also [docs](https://github.com/nvim-telescope/telescope.nvim#mappings). Third code snippet in this section.
For the previewer you wanna take a look [here](https://github.com/nvim-telescope/telescope.nvim/blob/4a8ea7763ec3c1c0672398418da682c1e0d4017d/lua/telescope/previewers.lua#L459-L483).

Also you have to carry the `tags_directory` with you so you wanna `return` it in this function as well